hhds/tree.hpp

```diff
@@ -71,111 +73,51 @@ class __attribute__((packed, aligned(64))) Tree_pointers {
     Tree_pos first_child_l              : CHUNK_BITS;
     Tree_pos last_child_l               : CHUNK_BITS;
 
+    // Track number of occupied short delta pointers
+    unsigned short num_short_del_occ    : CHUNK_SHIFT;
+
     // Short (delta) child pointers
-    Short_delta first_child_s_0         : SHORT_DELTA;
-    Short_delta first_child_s_1         : SHORT_DELTA;
-    Short_delta first_child_s_2         : SHORT_DELTA;
-    Short_delta first_child_s_3         : SHORT_DELTA;
-    Short_delta first_child_s_4         : SHORT_DELTA;
-    Short_delta first_child_s_5         : SHORT_DELTA;
-    Short_delta first_child_s_6         : SHORT_DELTA;
-
-    Short_delta last_child_s_0          : SHORT_DELTA;
-    Short_delta last_child_s_1          : SHORT_DELTA;
-    Short_delta last_child_s_2          : SHORT_DELTA;
-    Short_delta last_child_s_3          : SHORT_DELTA;
-    Short_delta last_child_s_4          : SHORT_DELTA;
-    Short_delta last_child_s_5          : SHORT_DELTA;
-    Short_delta last_child_s_6          : SHORT_DELTA;
+    __int128 first_child_s;
+    __int128 last_child_s;
 
     // Helper functions to get and set first child pointers by index
-    Short_delta _get_first_child_s(short index) const {
-        switch (index) {
-            case 0: return first_child_s_0;
-            case 1: return first_child_s_1;
-            case 2: return first_child_s_2;
-            case 3: return first_child_s_3;
-            case 4: return first_child_s_4;
-            case 5: return first_child_s_5;
-            case 6: return first_child_s_6;
-            
-            default: return INVALID;
-            // throw std::out_of_range("_get_first_child_s: Invalid index for first_child_s");
-        }
+    inline Short_delta _get_first_child_s(short index) const {
+        return (first_child_s >> (index * SHORT_DELTA)) & ((1 << SHORT_DELTA) - 1);
```

Comment:
The number of bits we use is 18 * 7 = 126. Thus, to us both unsigned and signed variants of `__int128` will work with no issues. This can definitely be addressed in future commits as a low priority task.